the-collab-lab/tcl-12-smart-shopping-list
# Issue #42: Restructure database and refactor code

## Description

- Refactored code so that all of API calls still work properly 

## Related Issue

Issue #42 


## Acceptance Criteria

- All API requests (getting, updating, deleting) work as before
- Filtering the list works as before
- Ordering the list works as before

## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|   | :sparkles: New feature     |
| ✓  | :hammer: Refactoring       |
|    | :100: Add tests            |
|    | :link: Update dependencies |
|    | :scroll: Docs              |

## Updates

### Before

![Screen Shot 2020-11-17 at 9 00 26 PM](https://user-images.githubusercontent.com/61370161/99473178-ae45f280-290f-11eb-96f6-83e9dd3d3ddd.png)


### After

![Screen Shot 2020-11-17 at 8 56 39 PM](https://user-images.githubusercontent.com/61370161/99473013-5e672b80-290f-11eb-9928-11f55afc4c35.png)

![Screen Shot 2020-11-17 at 8 56 49 PM](https://user-images.githubusercontent.com/61370161/99473024-62934900-290f-11eb-893d-03786a3901b4.png)


## Testing Steps / QA Criteria

- Switch to our branch (`git checkout al-eb-restructure-database`), then run `git pull origin al-eb-restructure-database`. Start server.
- Input the token "aft palsy buck"
- Open Firestore Console and find the corresponding token
- Items should appear in the app list in order of estimated days until next purchase (soonest, kind of soon, not soon, not purchased)
- Change the oldPurchased date in Firestore for an active item to a date that is more than calculatedEstimate x2 days before the lastPurchased date (if you need to, change calculatedEstimate to a lower number) to see it become inactive on the app
- Items should be sorted by soonest that needs to be purchased, as well as alphabetically if two items have the same number of days until next purchase
